### ISSUE TITLE
rkt does not work on latest k8s v1.0.6 on CoreOS

### ISSUE THREAD

**Issue by rimusz** [kubernetes/kubernetes#15638]
It looks like `rkt run` got changed and k8s cannot start pods, tried on latest CoreOS 833 and 766 builds:
````
kubectl describe pod my-nginx-kre94
Name:				my-nginx-kre94
Namespace:			default
Image(s):			nginx
Node:				172.19.17.99/
Labels:				run=my-nginx
Status:				Pending
Reason:				GeneralError
Message:			Query container info failed with error (cannot find status for pod: "my-nginx-kre94_default")
IP:
Replication Controllers:	my-nginx (2/2 replicas created)
Containers:
  my-nginx:
    Image:		nginx
    State:		Waiting
    Ready:		False
    Restart Count:	0
Events:
  FirstSeen				LastSeen			Count	From			SubobjectPath	Reason		Message
  Wed, 14 Oct 2015 09:39:09 -0600	Wed, 14 Oct 2015 09:39:09 -0600	1	{scheduler }				scheduled	Successfully assigned my-nginx-kre94 to 172.19.17.99
  Wed, 14 Oct 2015 09:39:09 -0600	Wed, 14 Oct 2015 09:39:50 -0600	6	{kubelet 172.19.17.99}			failedSync	Error syncing pod, skipping: invalid output: [NAME: 	image cat-manifest - Inspect and print the image manifest  USAGE: 	rkt image cat-manifest IMAGE  DESCRIPTION: 	IMAGE should be a string referencing an image; either a hash or an image name.  OPTIONS:   -h, --help=false: help for cat-manifest       --pretty-print=false: apply indent to format the output  GLOBAL OPTIONS:       --debug=false: Print out more debug information to stderr       --dir="/var/lib/rkt": rkt data directory       --insecure-skip-verify=false: skip all TLS, image or fingerprint verification       --local-config="/etc/rkt": local configuration directory       --system-config="/usr/lib/rkt": system configuration directory]
````

**Comment by dchen1107** [comment]
cc/ @yifan-gu

**Comment by yifan-gu** [comment]
@rimusz can you tell what's the version of rkt?

**Comment by rimusz** [comment]
I tried with v0.7 and v0.9

**Comment by yifan-gu** [comment]
@rimusz Current it only supports v0.8.1. But I will send a PR to fix for 0.9 as there are breaking changes in 0.9.

**Comment by rimusz** [comment]
@yifan-gu awesome, as I see it is a time to start using rkt with Kubernetes :)

**Comment by yifan-gu** [comment]
@rimusz yeah, FYI the network plugin piece and per container lifecycle hooks are still missing. Will work on the lifecycle hooks this week, but the network plugin needs to change its interface.

**Comment by yifan-gu** [comment]
Filed a fix for this at #15658

**Comment by rimusz** [comment]
@yifan-gu we at Deis are looking to switch to rkt and use v2 Deis under Kubernetes+rkt.
Is the network plugin interface can be fixed/updated soon too?

**Comment by yifan-gu** [comment]
@rimusz The problem is the interface takes a docker container ID, we need to change that to take a PID. But that involves many changes. I am curious what network plugin are you using? Currently by default the rkt pods on a node will run in one private subnet.

**Comment by rimusz** [comment]
@yifan-gu we are using the default network plugin.

**Comment by yifan-gu** [comment]
OK, if you are not specifying anything in the `--network-plugin` flag for kubelet, then the default one is no-op, so it won't affect you.

**Comment by rimusz** [comment]
no, I'm not specifying anything in the `--network-plugin` flag for kubelet. So that means we only need lifecycle hooks to be fixed then?

**Comment by yifan-gu** [comment]
@rimusz I think so.

**Comment by rimusz** [comment]
@yifan-gu cool then :)

**Comment by rimusz** [comment]
@yifan-gu have the lifecycle hooks got fixed?

**Comment by yifan-gu** [comment]
@rimusz I think the fix appears in v1.2.0-alpha.2
Still no lifecycle hooks yet, as the hooks defined in appc have small difference with the hooks defined in pod spec. No progress on further discussion yet. (I will bring this up in next rkt dev meeting)

FYI :https://github.com/kubernetes/kubernetes/issues/140#issuecomment-148548637
